### PR TITLE
Ninja and warnings in unit tests

### PR DESCRIPTION
**Pull request by jmarrec**
@Myoldmopar 

FOr some reason, these changes work in the unit test, but the GUI doesn't seem to register them. (I installed it by going to the root of the repo and typing `pip install -e .[dev,test]`)

**Edit**: Nevermind. I was selecting the build folder then clicking OK, I realize now that I needed to actually double click the build folder to enter it before clicking ok. The changes are still needed though.

### CODE REVIEW

File: epregressions/builds/base.py [hunk 1]
```diff
@@ -28,6 +28,8 @@ def autodetect_build_dir_type(build_dir: str) -> str:
                         return KnownBuildTypes.VisualStudio
                     elif 'Makefile' in generator_name:
                         return KnownBuildTypes.Makefile
+                    elif 'Ninja' in generator_name:
+                        return KnownBuildTypes.Makefile
```
Comment (jmarrec) [inline epregressions/builds/base.py:32]: Treat Ninja as Unix makefiles

File: epregressions/tests/builds/test_base.py [hunk 2]
```diff
@@ -33,6 +33,10 @@ def test_detect_visual_studio(self):
         self.add_cache_file('CMAKE_GENERATOR:INTERNAL=Visual Studio 2019')
         self.assertEqual(KnownBuildTypes.VisualStudio, autodetect_build_dir_type(self.build_dir))
 
+    def test_detect_ninja(self):
+        self.add_cache_file('CMAKE_GENERATOR:INTERNAL=Ninja')
+        self.assertEqual(KnownBuildTypes.Makefile, autodetect_build_dir_type(self.build_dir))
```
Comment (jmarrec) [inline epregressions/tests/builds/test_base.py:38]: test Detect ninja

File: epregressions/tests/builds/test_base.py [hunk 3]
```diff
@@ -58,11 +62,17 @@ def test_verify_without_setting_build_dir(self):
...
+        with open(os.path.join(temp_idf_dir, 'file2.iQQ'), 'w') as f:
+            f.write('he')
+        with open(os.path.join(temp_idf_dir, 'file3.idf'), 'w') as f:
+            f.write('ha')
+        with open(os.path.join(temp_idf_dir, 'file4.imf'), 'w') as f:
+            f.write('ha')  # macro
+        with open(os.path.join(temp_idf_dir, '_ExternalInterface-actuator.idf'), 'w') as f:
+            f.write('ha')  # ext interface as FMU
+        with open(os.path.join(temp_idf_dir, 'HVAC3ZoneGeometry.imf'), 'w') as f:
+            f.write('ha')  # macro resource file
```
Comment (jmarrec) [inline epregressions/tests/builds/test_base.py:76]: Avoid warning when running unittests `ResourceWarning: unclosed file <_io.TextIOWrapper`

**Review by jmarrec — COMMENTED** [review]
@Myoldmopar

File: epregressions/builds/base.py (hunk 1 above)
Comment (Myoldmopar) [reply epregressions/builds/base.py:32]: :+1:

File: epregressions/tests/builds/test_base.py (hunk 2 above)
Comment (Myoldmopar) [reply epregressions/tests/builds/test_base.py:38]: Nice.

File: epregressions/tests/builds/test_base.py (hunk 3 above)
Comment (Myoldmopar) [reply epregressions/tests/builds/test_base.py:76]: I guess could've appended .close() on each of the lines, but this is good.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
Merging this little thing in, sorry it took so long to context switch over to this, now quick I gotta go back to other work!

**Pull request merged**